### PR TITLE
add new package : nicstat

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/nicstat/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import os
+
+
+class Nicstat(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nicstat/package.py]: This should be `MakefilePackage` base class.

File: var/spack/repos/builtin/packages/nicstat/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    """
+
+    homepage = "https://github.com/scotte/nicstat"
+    url      = "https://jaist.dl.sourceforge.net/project/nicstat/nicstat-1.95.tar.gz"
+
+    version('1.95', sha256='c4cc33f8838f4523f27c3d7584eedbe59f4c587f0821612f5ac2201adc18b367')
+
+    def install(self, spec, prefix):
+        os.popen("mv Makefile.Linux makefile && sed -i '23cCMODEL=""' makefile\
+        && sed -i 's/sudo/''/g' makefile && mkdir /usr/local/share/man/man1 ")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nicstat/package.py]: We'll need to split this up into Spack/Python commands. Spack plans on supporting Windows in the future, so you can't assume these commands exist on the system. Also, `sed -i` behaves differently on macOS. For example, you can do:
```python
copy('Makefile.Linux', 'makefile')
filter_file('23cCMODEL=""', '', 'makefile', string=True)
filter_file('sudo', '', 'makefile', string=True)
```
Also, Spack shouldn't install anything outside of the installation prefix, so you shouldn't need to create that last directory.

File: var/spack/repos/builtin/packages/nicstat/package.py (hunk 1 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/nicstat/package.py]: fix this in https://github.com/spack/spack/pull/14173/commits/14b6d91a65b36d3a1155901985eae15554ecec3a

File: var/spack/repos/builtin/packages/nicstat/package.py (hunk 2 above)
Comment (darmac) [reply var/spack/repos/builtin/packages/nicstat/package.py]: fix this in https://github.com/spack/spack/pull/14173/commits/14b6d91a65b36d3a1155901985eae15554ecec3a

File: var/spack/repos/builtin/packages/nicstat/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    """
+
+    homepage = "https://github.com/scotte/nicstat"
+    url      = "https://jaist.dl.sourceforge.net/project/nicstat/nicstat-1.95.tar.gz"
+
+    version('1.95', sha256='c4cc33f8838f4523f27c3d7584eedbe59f4c587f0821612f5ac2201adc18b367')
+
+    def edit(self, spec, prefix):
+        copy('Makefile.Linux', 'makefile')
+        filter_file('CMODEL =\\s+-m32', '', 'makefile')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/nicstat/package.py]: You can use `r'...'` for raw formatting so you don't need the double slash:
```suggestion
        filter_file(r'CMODEL =\s+-m32', '', 'makefile')
```

Comment (darmac) [reply var/spack/repos/builtin/packages/nicstat/package.py]: Thanks very much for your guide, I fix it in this patch : https://github.com/spack/spack/pull/14173/commits/aacb061bc58bbf686b375c741690d336569274df

**Comment by adamjstewart** [comment]
Closing and reopening to try to fix Travis failures.

**Pull request merged**